frontend/src/container/GridCardLayout/WidgetHeader/index.tsx

```diff
@@ -75,15 +81,34 @@ function WidgetHeader({
 
 	const onCreateAlertsHandler = useCreateAlerts(widget);
 
+	const onDownloadHandler = useCallback((): void => {
+		const csv = unparse(tableProcessedDataRef.current);
```

Comment:
We should add error handling here

Reply:
will add everywhere this function is called in next PR